pymc/gp/gp.py

```diff
@@ -131,9 +130,9 @@ def _build_prior(self, name, X, reparameterize=True, jitter=JITTER_DEFAULT, **kw
         mu = self.mean_func(X)
         cov = stabilize(self.cov_func(X), jitter)
         if reparameterize:
-            size = infer_size(X, kwargs.pop("size", None))
+            size = np.shape(X)[0]
             v = pm.Normal(name + "_rotated_", mu=0.0, sigma=1.0, size=size, **kwargs)
-            f = pm.Deterministic(name, mu + cholesky(cov).dot(v))
+            f = pm.Deterministic(name, mu + cholesky(cov).dot(v), dims=kwargs.pop("dims", None))
```

Comment:
No, but it also makes no difference.
Since I don't have the branch in front of me, could we go ahead with the current version?
Seems overkill to rebase, push and run the CI for this..

Unless you have another suggestion, of course

Reply:
I think its just `pop` will mutate kwargs and `get` wont.  I'm in favor of `get`, because `v` right above `f` should also be changed to also get passed `dims` if the user provides it.

Reply:
The `Normal` already gets the `dims`, because if they're passed they will be in the `**kwargs`.
Mutating the kwargs won't matter, because the method ends in the line after.

But I can change it if it makes you feel better ;)

Reply:
Ah gotcha, if both `f` and `v` get dims then I feel great

Reply:
There's no reason to pop, even if right now the function call ends here. It might be extended in the future and devs will have to investigate if there was any reason for the pop in the first place.

Reply:
I changed it to a `.get()` and added a commit to pin the blackjax version to fix the CI.